src/C_API.jl

```diff
@@ -182,6 +184,32 @@ function _c_jacobian_evaluation(
     return err
 end
 
+function _c_variable_name(id_ptr::Ptr{Cvoid}, variable_idx::Cint, buffer::Ptr{Cuchar}, buf_size::Cint)
+    id_data = unsafe_pointer_to_objref(id_ptr)::InterfaceData
+    str = id_data.variable_name[variable_idx]
+    # mimicking `strncpy` of Clang
+    if length(str) < buf_size
+        str *= "\0" ^ (buf_size-length(str))
+    end
+    str = str[1:buf_size] * "\0"
+    unsafe_copyto!(buffer, pointer(str), length(str))
+    return
+end
```

Comment:
Did you copy this from somewhere? Two questions:
* Is `buf_size` meant to include the final `\0`? Or is that meant to be at `buf_size+1`?
* Did you try this with unicode? `length(str)` is not necessarily the same is the length of the `codeunits(str)`
* You should be careful whenever you call `pointer(str)`. It's probably okay in this case, but we should probably wrap it in a `GC.@preserve`.

I haven't tested, but maybe something like:
```Julia
function _c_variable_name(
    id_ptr::Ptr{Cvoid},
    i::Cint,
    buf_ptr::Ptr{UInt8},
    buf_size::Cint,
)
    id_data = unsafe_pointer_to_objref(id_ptr)::InterfaceData
    name = collect(codeunits(id_data.variable_name[i]))
    n = length(name)
    resize!(name, buf_size + 1)
    name[min(buf_size, n):end] .= UInt8('\0')
    GC.@preserve name begin
        unsafe_copyto!(buf_ptr, pointer(name), buf_size)
    end
    return
end
```

Reply:
I thought I copied from PATH examples, but can't find.  I used similar lines in [the old interface](https://github.com/chkwon/PathJulia/blob/master/src/pathjulia.c#L82-L92) for PATHSolver.jl 

I actually don't understand how `buf_size` actually works within PATH. It seems no manual truncation of the string is necessary as `unsafe_copyto!` does it anyway. Plus, PATH seems also truncate it internally also. But not 100% sure. I kept the manual truncation though, just in case.